src/lib/sse/CustomEventSource.ts

```diff
@@ -11,13 +14,40 @@ const CONNECTION_CLOSED = 2
 const DEFAULT_RECONNECT_INTERVAL = 1000
 const MAX_RETRIES = 3
 
-export default class CustomEventSource {
-  public esOnError: (err: any) => void
-  public esOnMessage: () => void
-  public esOnOpen: () => void
+interface EventListeners {
+  event: string
+  handler: any
+}
+
+type OnMessageHandler = (data: any) => void
+type OnErrorHandler = (err: EventSourceError) => void
+type OnOpenHandler = () => void
+
+export class CustomEventSource {
+  public static create(source: string, closeOnInvalidToken = false) {
+    let token
+    if (closeOnInvalidToken) {
+      token = SessionManager.getSessionManager().checkAndGetToken(closeOnInvalidToken)
+    } else {
+      token = SessionManager.getSessionManager().token
+    }
+
+    return new CustomEventSource(source, {
+      headers: {
+        authorization: `bearer ${token}`,
+        'user-agent': userAgent,
+        ...(envCookies() ? { cookie: envCookies() } : null),
+        ...(cluster() ? { 'x-vtex-upstream-target': cluster() } : null),
```

Comment:
Just an idea as an improvement (not necessary for this PR): we could make a const object `VTEX_HEADERS` and save the headers as constants there. Like this:
```
const VTEX_HEADERS = {
   UPSTREAM_TARGET = 'x-vtex-upstream-target',
   ...
}
```

Reply:
Really nice idea!!!! In the future we definitely need this